python/cugraph/structure/graph.py

```diff
@@ -1113,10 +1130,14 @@ def has_edge(self, u, v):
         if self.edgelist is None:
             raise Exception("Graph has no Edgelist.")
         if self.renumbered:
-            tmp = self.renumber_map.to_internal_vertex_id(cudf.Series([u, v]))
+            tmp = cudf.DataFrame({'src': [u, v]})
+            tmp = tmp.astype({'src': 'int'})
+            tmp = self.add_internal_vertex_id(
+                tmp, 'id', 'src', preserve_order=True
+            )
 
-            u = tmp[0]
-            v = tmp[1]
+            u = tmp['id'][0]
+            v = tmp['id'][1]
 
         df = self.edgelist.edgelist_df
         if self.distributed:
```

Comment:
Clearly outside the scope of this PR, but we need to inspect the performance of this approach. Ideally, we can figure out which GPU can possibly hold (u, v), and in the target GPU, we just need to check ```u```'s adjacency list (if store_transposed = False). The code below sounds much more expensive than this (and does this assume availability of COO?, are we storing COO even after graph creation? This more than halves the maximum graph size we can handle)

Reply:
We will need a new version of `has_edge`.  Do we want to create an issue for that specifically, or will we naturally do that as we test the python graph object in MNMG mode?  Not sure if we want to create a large number of little issues as we see them or trust that when we try to use it we'll notice :-)

Reply:
Yeah... just ignore this comment. It's clearly outside the scope, but just wanted to raise this issue to make sure more people aware of the problem.